### DISCUSSION TITLE
Pass a stable and random id from Server to Client Component?

### DISCUSSION THREAD

**Discussion by tinleym** [vercel/next.js#44601] [Help] [answered]
I'd like to generate a stable and random id in a server component that can be passed to a client component. The client approach, useId(), isn't compatible. Since (to my understanding) server components are stable per route loaded, would a simple crypto.randomUUID() work?

**Comment by MoltenCoffee** [comment, marked as answer]
I think it all depends on the meaning of "stable". What do you consider stable in this case?

**Reply by tinleym** [reply]
I mean generated once and remaining the same until a page refresh. 

For example, if I assign a random id to a dom node in a Server Component, I'd like to be able to reliably reference that id in a Client Component.

**Reply by MoltenCoffee** [reply]
Then yes, as far as I know you should be able to. Server components do not have React state and lifecycle, so any data passed should should be static, if I understand correctly.

But I would try it out if I were you!

**Reply by tinleym** [reply]
Thank you! It is working for me so far... I was just wondering if anyone could think of a way for it to break.

**Comment by GabenGar** [comment]
There is a [`useId()` hook](https://reactjs.org/docs/hooks-reference.html#useid) in React, which sounds like something suited for your needs.

**Reply by tinleym** [reply]
useId won't work in a server component
